integration_tests/test_pv_connection.py

```diff
@@ -25,26 +25,26 @@ def on_market_cycle(self, market_info):
                 offer_info = json.loads(offer["offer"])
                 assert offer_info["price"] == 50 * market_info["available_energy_kWh"]
                 assert offer_info["energy"] == market_info["available_energy_kWh"]
-                # Validate that the bid was placed to the market
+                # Validate that the offer was placed to the market
                 offer_listing = self.list_offers()
-                listed_offer = next(bid for bid in offer_listing["offer_list"] if bid["id"] == offer_info["id"])
+                listed_offer = next(offer for offer in offer_listing["offer_list"] if offer["id"] == offer_info["id"])
                 assert listed_offer["price"] == listed_offer["price"]
                 assert listed_offer["energy"] == listed_offer["energy"]
-                # Try to delete the bid
+                # Try to delete the offer
                 delete_resp = self.delete_offer(offer_info["id"])
                 assert delete_resp["deleted_offer"] == offer_info["id"]
-                # Validate that the bid was deleted from the market
-                empty_listing = self.list_bids()
+                # Validate that the offer was deleted from the market
+                empty_listing = self.list_offers()
                 assert not any(o for o in empty_listing["offer_list"] if o["id"] == offer_info["id"])
-                # Place the bid with a price that will be acceptable for trading
+                # Place the offer with a price that will be acceptable for trading
                 offer = self.offer_energy(market_info["available_energy_kWh"], 10 * market_info["available_energy_kWh"])
                 offer_info = json.loads(offer["offer"])
                 assert offer_info["price"] == 10 * market_info["available_energy_kWh"]
                 assert offer_info["energy"] == market_info["available_energy_kWh"]
 
             stats = self.list_stats()
             traded_slots = stats["market_stats"]["energy_trade_profile"]["sold_energy"]["pv"]["accumulated"].values()
-            assert isclose(stats["device_stats"]["bills"]["bought"], sum(traded_slots))
+            assert isclose(stats["device_stats"]["bills"]["sold"], sum(traded_slots), rel_tol=0.002)
```

Comment:
Do you think at some point we should also introduce global constant for `rel_tol` too as its value is different over all our repos. ?

Reply:
Maybe, but this is not the case here. The reason why the tolerance has such a big value (0.002 is rather big IMO) is because the results that we read from d3a are in a 3 decimal point precision. Therefore, there might be a deviation of 0.001 due to rounding. That's why the tolerance has to be at least 0.002 in order for all the checks to pass.